queries.sql

```diff
@@ -57,4 +57,201 @@ GROUP BY species;
+
+SELECT species, AVG(escape_attempts) as avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
+
+
+SELECT * FROM animals;
+
+CREATE TABLE owners (
+    id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    full_name VARCHAR(100),
+    age INTEGER
+);
+CREATE TABLE species (
+    id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(50)
+);
+
+ALTER TABLE animals
+    DROP COLUMN species,
+    ADD COLUMN species_id INTEGER,
+    ADD COLUMN owner_id INTEGER,
+    MODIFY COLUMN id INTEGER NOT NULL AUTO_INCREMENT;
+
+ALTER TABLE animals
+    ADD CONSTRAINT fk_species_id FOREIGN KEY (species_id)
+        REFERENCES species(id),
+    ADD CONSTRAINT fk_owner_id FOREIGN KEY (owner_id)
+        REFERENCES owners(id);
+
+-- find owner_ids
+SELECT id, full_name FROM owners;
+
+  SELECT animals.name
+FROM animals
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.visit_date DESC
+LIMIT 1;
+
+SELECT COUNT(DISTINCT visits.animal_id)
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Stephanie Mendez';
+
+SELECT vets.name, species.name AS specialty
+FROM vets
+LEFT JOIN specializations ON vets.id = specializations.vet_id
+LEFT JOIN species ON specializations.species_id = species.id
+ORDER BY vets.name, specialty;
+
+
+
+SELECT animals.name
+FROM animals
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Stephanie Mendez' AND visits.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT animals.name, COUNT(visits.id) AS num_visits
+FROM animals
+JOIN visits ON animals.id = visits.animal_id
+GROUP BY animals.id
+ORDER BY num_visits DESC
+LIMIT 1;
+
+SELECT vets.name, animals.name, MIN(visits.visit_date) AS first_visit
+FROM vets
+JOIN visits ON vets.id = visits.vet_id
+JOIN animals ON visits.animal_id = animals.id
+WHERE vets.name = 'Maisy Smith'
+GROUP BY vets.name, animals.name
+ORDER BY first_visit ASC
+LIMIT 1;
+
+SELECT animals.name AS animal_name, vets.name AS vet_name, visits.visit_date
+FROM visits
+INNER JOIN animals ON visits.animal_id = animals.id
+INNER JOIN vets ON visits.vet_id = vets.id
+ORDER BY visits.visit_date DESC
+LIMIT 1;
+
+SELECT COUNT(*) AS num_visits
+FROM visits
+INNER JOIN animals ON visits.animal_id = animals.id
+INNER JOIN vets ON visits.vet_id = vets.id
+INNER JOIN specializations ON vets.id = specializations.vet_id
+INNER JOIN species ON specializations.species_id = species.id
+WHERE species.id != animals.species_id;
+
+SELECT species.name, COUNT(*) AS num_visits
+FROM visits
+INNER JOIN animals ON visits.animal_id = animals.id
+INNER JOIN vets ON visits.vet_id = vets.id
+INNER JOIN specializations ON vets.id = specializations.vet_id
+INNER JOIN species ON specializations.species_id = species.id
+WHERE vets.name = 'Maisy Smith'
+GROUP BY species.name
+ORDER BY num_visits DESC
+LIMIT 1;
```

Comment:
- Please add a screenshot of this query, since there´s no result for this one. It should give you a result with the name of Maisy Smith, the species, and the count of the species she get the most